### PR TITLE
fix:navigation bar will not follow scroll bar scroll

### PR DESCRIPTION
**Pull request by MapleWithoutWords**
https://user-images.githubusercontent.com/38368335/226389350-08453bed-5cab-4c56-bdcd-357f45014c71.mp4

### CODE REVIEW

File: docs/Masa.Docs.Shared/Pages/Index.razor [hunk 1]
```diff
@@ -27,28 +28,38 @@
...
-                {
-                    <MListItem OnClick="(() => ScrollIntoView(project))" Value="project" @key="project">
-                        <MListItemAvatar Tile>
-                            @Image(project, _active)
-                        </MListItemAvatar>
-
+            <ul style="list-style-type: none;padding:0px !important;">
+                <li>
+                    <a href="@($"#getting-started")" class="m-list-item m-list-item--link @GenClass("getting-started")"
+                       @onclick:preventDefault
```
Comment (doddgu) [inline docs/Masa.Docs.Shared/Pages/Index.razor:34]: format code

File: docs/Masa.Docs.Shared/Pages/Index.razor [hunk 2]
```diff
@@ -27,28 +28,38 @@
...
+                            <MListItemTitle>@I18n.T("getting-started")</MListItemTitle>
                         </MListItemContent>
-                    </MListItem>
+                    </a>
+                </li>
+                @foreach (var (project, navs) in _projectToNavs.OrderByDescending(u => u.Key))
+                {
+                    <li>
+                        <a href="@($"#{project}")"
+                       class="m-list-item m-list-item--link @GenClass(project)"
```
Comment (doddgu) [inline docs/Masa.Docs.Shared/Pages/Index.razor:48]: format code

**Review by doddgu — CHANGES_REQUESTED** [review]

File: docs/Masa.Docs.Shared/wwwroot/js/docs.js [hunk 3]
```diff
@@ -234,24 +234,25 @@ window.MasaBlazor.markdownItRules = function (parser) {
   }
 };
 
-window.registerWindowScrollEventForToc = function (dotnet, tocId) {
+window.registerWindowScrollEvent = function (dotnet, className) {
```
Comment (capdiem) [inline docs/Masa.Docs.Shared/wwwroot/js/docs.js:237]: 直接传 cssSelector 不就简单了吗？

File: docs/Masa.Docs.Shared/Shared/Toc.razor [hunk 4]
```diff
@@ -27,7 +27,7 @@
         <ul class="mb-6">
             @foreach (var item in _toc)
             {
-                <li class="pl-3 text-body-2 py-1 font-weight-regular @GenClass(item)">
+                <li class="pl-3 text-body-2 py-1 font-weight-regular @GenClass(item) scroll-change-hash">
```
Comment (capdiem) [inline docs/Masa.Docs.Shared/Shared/Toc.razor:30]:
```suggestion
                <li class="pl-3 text-body-2 py-1 font-weight-regular @GenClass(item) toc-li">
```

File: docs/Masa.Docs.Shared/Pages/Index.razor [hunk 5]
```diff
@@ -27,20 +28,20 @@
...
-            <MListItemGroup @bind-Value="_active" Mandatory>
-                <MListItem OnClick='() => ScrollIntoView("getting-started")'>
-                    <MListItemIcon Style="margin-left: 6px;margin-right: 18px;">
-                        <MIcon Color="#FFB547" Size="30" Class="mr-2">fas fa-rocket</MIcon>
-                    </MListItemIcon>
-                    <MListItemContent>
-                        <MListItemTitle>@I18n.T("getting-started")</MListItemTitle>
-                    </MListItemContent>
+            <MListItemGroup Value="_activeHash" Mandatory>
+                <MListItem Value="@("#getting-started")" Class="scroll-change-hash" OnClick='() => ScrollIntoView("getting-started")' Attributes="@(new Dictionary<string,object>{ {"href" , "#getting-started" }})">
```
Comment (capdiem) [inline docs/Masa.Docs.Shared/Pages/Index.razor:32]:
```suggestion
                <MListItem Value="@("#getting-started")" Class="doc-list-nav--item-hash" OnClick='() => ScrollIntoView("getting-started")' Attributes="@(new Dictionary<string,object>{ {"href" , "#getting-started" }})">
```

**Review by capdiem — CHANGES_REQUESTED** [review]

**Review by capdiem — APPROVED** [review]

**Pull request merged**